pepper-apis/src/main/java/org/broadinstitute/ddp/route/ReceiveDsmNotificationRoute.java

```diff
@@ -0,0 +1,86 @@
+package org.broadinstitute.ddp.route;
+
+import org.apache.http.HttpStatus;
+import org.broadinstitute.ddp.constants.ErrorCodes;
+import org.broadinstitute.ddp.constants.RouteConstants.PathParam;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.db.dao.JdbiUserStudyEnrollment;
+import org.broadinstitute.ddp.db.dto.StudyDto;
+import org.broadinstitute.ddp.json.dsm.DsmNotificationPayload;
+import org.broadinstitute.ddp.json.errors.ApiError;
+import org.broadinstitute.ddp.model.activity.types.DsmNotificationEventType;
+import org.broadinstitute.ddp.model.event.DsmNotificationSignal;
+import org.broadinstitute.ddp.model.user.EnrollmentStatusType;
+import org.broadinstitute.ddp.model.user.User;
+import org.broadinstitute.ddp.service.EventService;
+import org.broadinstitute.ddp.util.ResponseUtil;
+import org.broadinstitute.ddp.util.RouteUtil;
+import org.broadinstitute.ddp.util.ValidatedJsonInputRoute;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+/**
+ * A hook called by DSM at a certain stage, triggers the notification sending when invoked
+ * Given a user GUID, a study GUID and a notification event type, looks for a notification template
+ * and queues an event which is picked up by Housekeeping later and results in sending the notification
+ * Returns:
+ *    200 OK - the notification event was queued or no event configuration exists
+ *    404 Not Found - no study, user or notification event type found
+ */
+public class ReceiveDsmNotificationRoute extends ValidatedJsonInputRoute<DsmNotificationPayload> {
```

Comment:
General comment:
May be existing "DsmTriggerOnDemandActivityRoute" (creates followup activity instances) can be replaced with this "ReceiveDsmNotificationRoute" probably add `activity_instance_creation` EventTriggerType and handle it in "ReceiveDsmNotificationRoute" .
One less route to maintain and cleaner. Probably need DSM change . May be future TechDebt tkt !!!

Reply:
It will require changes on DSM side. Not going to do now.

Reply:
I agree, thats the reason I mentioned may be future TechDebt. We can deal with it later.